src/lib.rs

```diff
@@ -1,14 +1,21 @@
 extern crate serde_json;
 
-mod data;
-mod expression;
-mod operators;
+pub mod data;
+pub mod expression;
+pub mod operators;
 
+use expression::Ambiguous;
 use serde_json::Value;
 use std::collections::HashSet;
 
 use data::Data;
 
+#[derive(Debug, PartialEq)]
+pub enum PartialApplyResult {
```

Comment:
`nitpick`
```suggestion
pub enum PartialApplyOutcome {
```
Just so that it doesn't say `Result<PartialApplyResult, T>`